src/main/scala/shark/memstore2/MemoryMetadataManager.scala

```
@@ -54,10 +54,16 @@ class MemoryMetadataManager extends LogHelper {
       databaseName: String,
       tableName: String,
       cacheMode: CacheType.CacheType): MemoryTable = {
-    val tableKey = MemoryMetadataManager.makeTableKey(databaseName, tableName)
-    val newTable = new MemoryTable(databaseName, tableName, cacheMode)
-    _tables.put(tableKey, newTable)
-    newTable
+	val tableKey = MemoryMetadataManager.makeTableKey(databaseName, tableName)
+    // Clear out any existing tables with the same key; prevent memory leak
+    if (containsTable(databaseName, tableName)) {
+      logInfo("Attempt to create new table when one already exists - " + tableKey)
```

We could silently ignore repeated cache calls, or return a more descriptive error message saying that the first cache call insures that successive inserts on a `MEMORY` table will be pinned in memory.